arborist/user.go

```diff
@@ -87,14 +87,31 @@ func userWithName(db *sqlx.DB, name string) (*UserFromQuery, error) {
 			usr.email,
 			array_remove(array_agg(DISTINCT grp.name), NULL) AS groups,
 			(
-				SELECT json_agg(json_build_object('policy', policy.name, 'expires_at', usr_policy.expires_at))
-				FROM usr_policy
-				INNER JOIN policy ON policy.id = usr_policy.policy_id
-				WHERE usr_policy.usr_id = usr.id
+				SELECT json_agg(json_build_object('policy', all_policies.name, 'expires_at', all_policies.expires_at))
+				FROM (
+					SELECT policy.name AS name, usr_policy.expires_at AS expires_at
+					FROM usr_policy
+					INNER JOIN policy ON policy.id = usr_policy.policy_id
+					WHERE usr_policy.usr_id = usr.id
+					UNION
+					SELECT policy.name AS name, usr_grp.expires_at AS expires_at
+					FROM usr_grp
+					INNER JOIN grp_policy ON grp_policy.grp_id = usr_grp.grp_id
+					INNER JOIN policy ON policy.id = grp_policy.policy_id
+					WHERE usr_grp.usr_id = usr.id
+					UNION
+					SELECT policy.name AS name, NULL AS expires_at
+					FROM grp
+					INNER JOIN grp_policy ON grp_policy.grp_id = grp.id
+					INNER JOIN policy ON policy.id = grp_policy.policy_id
+					WHERE grp.name IN ('anonymous', 'logged-in')
+				) AS all_policies
 			) AS policies
 		FROM usr
 		LEFT JOIN usr_grp ON usr_grp.usr_id = usr.id
-		LEFT JOIN grp ON grp.id = usr_grp.grp_id
+		LEFT JOIN grp ON (
+			grp.id = usr_grp.grp_id OR grp.name IN ('anonymous', 'logged-in')
```

Comment:
In this case, there's a special case in the last UNION block, where we need to return NULL as the expiration date (policies assigned to the anonymous and loggedin groups don't expire, and users' membership in the anonymous and loggedin groups also doesn't expire).

Reply:
note -> we're doing `SELECT policy.name AS name, NULL AS expires_at` for anonymous and logged-in groups because there's no expiration for the user being in these groups, while we get the real expiration for other groups

Reply:
I'm going to add a comment to the code too!